docs/sdkx-cocos2dx/tracking.mdx

```diff
@@ -0,0 +1,215 @@
+
+All the public APIs in the SDK should be called after initializing the SDK via [install API](/sdkx-cocos2dx/getting-started#initializing)
+
+</Admonition>
+
+## Adding Tags to Conversations {#adding-tags}
+
+You can attach tags while reporting an issue by passing them to the `configMap` object at the time of calling `HelpshiftCocos2dx::showConversation(configMap)`. You can pass an **array of strings** with a key `"tags"` which will get added as Tags when the issue is created.
+
+For example:
+
+```cpp
+cocos2d::ValueMap getConfig() {
+    cocos2d::ValueMap configMap;
+
+    ValueVector tags;
+    tags.push_back(Value("foo"));
+    tags.push_back(Value("bar"));
+
+    configMap["tags"] = tags;
+    return configMap;
+}
+
+cocos2d::ValueMap chatConfig = getConfig();
+HelpshiftCocos2dx::showConversation(chatConfig);
+
+```
+
+<Admonition type="info" title="Note">
+
+**On tag names & compatibility**
+
+- Tags must be pre-created in the Helpshift Dashboard (Settings → Tags), otherwise they will be ignored.
+- The attached tags must exactly match the tags present on the dashboard.
+
+</Admonition>
+
+## Adding Custom Metadata to Conversations
+
+You can attach additional metadata to every new conversation started by the app user. This metadata can include properties like username, email, game scores, current game levels, and any other data needed to provide relevant context for each new conversation. You can attach custom metadata by passing it to the `configMap` object at the time of calling any of the SDK X APIs (`showConversation`, `showFAQs`, `showSingleFAQ`, `showFAQSection`).
+
+Example:
+
+```cpp
+ValueMap config;
+ValueMap meta;
+meta["usertype"] = Value("paid");
+meta["level"] = Value("7");
+meta["score"] = Value("12345");
+
+config["customMetadata"] = Value(meta);
+
+HelpshiftCocos2dx::showFAQs(config);
+```
+
+<Admonition type="info" title="Note">
+
+- Metadata should only be sent as String key-value pairs.
+- Once customMetadata is set, if you want to reset it, then you’ll have to call the SDK X API with an empty map, i.e.
+
+```cpp
+ValueMap config;
+ValueMap meta;
+config["customMetadata"] = Value(meta);
+HelpshiftCocos2dx::showFAQs(config);
+```
+
+</Admonition>
+
+## Attaching Custom Issue Fields to conversations {#set-custom-issue-fields}
+
+You can attach Custom Issue Fields to every new conversation started by the user. A Custom Issue Field should have a key, a datatype, and a value. The Helpshift SDK allows the addition of Custom Issue Fields by using the `cifs` key in API config map.
+
+<Admonition type="info" title="Note">
+  We would like to inform you that the usage of the <b>"customIssueFields"</b>{" "}
+  key will be deprecated. We strongly recommend transitioning to using the{" "}
+  <b>"cifs"</b> key as the preferred method for passing custom issue fields.
+</Admonition>
+
+These custom issue fields will be sent whenever an end user starts a new conversation or creates an issue via Smart FAQs.
```

Comment:
why is this line needed for Smart FAQs specially ?

Reply:
It's the same way in all docs.